letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
+
             # Don't send challenges for None and False authenticator responses
-            if resp:
+            if resp is not None and resp:
```

Comment:
For the same reason we use `if something is not None` rather than `if something`. This way it's clear that `resp` could actually be `None`, which is in fact important considering API (description in `interfaces`).